src/engine/enginemaster.cpp

```diff
@@ -443,48 +485,213 @@ void EngineMaster::process(const int iBufferSize) {
+
+        MicMonitorMode configuredMicMonitorMode = static_cast<MicMonitorMode>(
+            static_cast<int>(m_pMicMonitorMode->get()));
+
+        // Process master, booth, and record/broadcast buffers according to the
+        // MicMonitorMode configured in DlgPrefSound
+        if (configuredMicMonitorMode == MicMonitorMode::MASTER) {
+            // Process master channel effects
+            // TODO(Be): Move this after mixing in talkover. To apply master effects
+            // to both the master and booth in that case will require refactoring
+            // the effects system to be able to process the same effects on multiple
+            // buffers within the same callback.
+            if (m_pEngineEffectsManager) {
+                GroupFeatureState masterFeatures;
+                // Well, this is delayed by one buffer (it's dependent on the
+                // output). Oh well.
+                if (m_pVumeter != NULL) {
+                    m_pVumeter->collectFeatures(&masterFeatures);
+                }
+                masterFeatures.has_gain = true;
+                masterFeatures.gain = m_pMasterGain->get();
+                m_pEngineEffectsManager->process(m_masterHandle.handle(), m_pMaster,
+                                                iBufferSize, iSampleRate,
+                                                masterFeatures);
+            }
+
+            // Copy master mix to booth output with booth gain before mixing
+            // talkover with master mix
+            if (boothEnabled) {
+                CSAMPLE boothGain = m_pBoothGain->get();
+                if (m_bRampingGain) {
+                    SampleUtil::copy1WithRampingGain(m_pBooth, m_pMaster,
+                        m_boothGainOld, boothGain, iBufferSize);
+                } else {
+                    SampleUtil::copy1WithGain(m_pBooth, m_pMaster,
+                        boothGain, iBufferSize);
+                }
+                m_boothGainOld = boothGain;
+            }
+
+            // Mix talkover into master mix
+            if (m_pNumMicsConfigured->get() > 0) {
+                SampleUtil::copy2WithGain(m_pMaster,
+                    m_pMaster, 1.0,
                     m_pTalkover, 1.0,
                     iBufferSize);
-        } else {
-            SampleUtil::copy3WithGain(m_pMaster,
-                    m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+            }
+
+            // Apply master gain
+            CSAMPLE master_gain = m_pMasterGain->get();
+            if (m_bRampingGain) {
+                SampleUtil::applyRampingGain(m_pMaster, m_masterGainOld, master_gain,
+                                             iBufferSize);
+            } else {
+                SampleUtil::applyGain(m_pMaster, master_gain, iBufferSize);
+            }
+            m_masterGainOld = master_gain;
```

Comment:
Hmm, it could come in handy to make a static SampleUtil function like:
`SampleUtil::ramp(CSAMPLE* pBuffer, CSAMPLE_GAIN* oldGain, CSAMPLE* newGain)`
that applies the ramping gain to the buffer and updates the oldGain variable with the newGain value afterward. Any suggestions for a better function name?